### PR TITLE
Clean up build tasks

### PR DESCRIPTION
**Pull request by tmat**
Removes unnecessary dependencies from SignTool task.
Use eng/BuildTask.targets in SignTool project.
Avoid publishing when building for desktop fx. This is unnecessary and causes double-writes.
Do not include assemblies that ship with MSBuild in task packages (specifically System.Collections.Immutable).

### CODE REVIEW

**Comment by tmat** [comment]
@weshaggard PTAL

File: eng/BuildTask.targets [hunk 1]
```diff
@@ -27,45 +27,57 @@
...
+
+    <!-- 
+      Do not include assemblies that MSBuild ships with in the package.
+    -->
+    <PackageReference Update="Microsoft.Build" PrivateAssets="All" Publish="false" ExcludeAssets="runtime" />
+    <PackageReference Update="Microsoft.Build.Framework" PrivateAssets="All" Publish="false" ExcludeAssets="runtime" />
+    <PackageReference Update="Microsoft.Build.Tasks.Core" PrivateAssets="All" Publish="false" ExcludeAssets="runtime" />
+    <PackageReference Update="Microsoft.Build.Utilities.Core" PrivateAssets="All" Publish="false" ExcludeAssets="runtime" />
+    <PackageReference Update="System.Collections.Immutable" PrivateAssets="All" Publish="false" ExcludeAssets="runtime" />
+    <PackageReference Update="System.IO.Compression" PrivateAssets="All" Publish="false" ExcludeAssets="runtime" />
```
Comment (weshaggard) [inline eng/BuildTask.targets]: What does update do if it isn't included already? Does it add it or does it ignore it?

File: src/Microsoft.DotNet.Arcade.Sdk/src/AssemblyResolution.cs [hunk 2]
```diff
@@ -21,13 +21,26 @@ private static Assembly AssemblyResolve(object sender, ResolveEventArgs args)
...
+                sci = Assembly.LoadFile(fullPath);
+            }
+            catch
+            {
+                return null;
+            }
+
+            if (name.Version <= sci.GetName().Version)
+            {
+                return sci;
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Arcade.Sdk/src/AssemblyResolution.cs]: Is this even necessary? Are there any versions of msbuild that don't carry System.Collections.Immutable? I know at least 15.7+ carry it.

File: eng/BuildTask.targets (hunk 1 above)
Comment (tmat) [reply eng/BuildTask.targets]: Ignore.

File: src/Microsoft.DotNet.Arcade.Sdk/src/AssemblyResolution.cs [hunk 3]
```diff
@@ -21,13 +21,26 @@ private static Assembly AssemblyResolve(object sender, ResolveEventArgs args)
...
+            try
+            {
+                sci = Assembly.LoadFile(fullPath);
+            }
+            catch
+            {
+                return null;
+            }
+
+            if (name.Version <= sci.GetName().Version)
```
[the remark this answers is not in the document]
Comment (tmat) [reply src/Microsoft.DotNet.Arcade.Sdk/src/AssemblyResolution.cs]: It doesn't return null.

File: src/Microsoft.DotNet.Arcade.Sdk/src/AssemblyResolution.cs (hunk 2 above)
Comment (tmat) [reply src/Microsoft.DotNet.Arcade.Sdk/src/AssemblyResolution.cs]: They do. But SRM references version 1.2.0, while the version in MSBuild 15.7+ is 1.2.3.

Comment (tmat) [reply src/Microsoft.DotNet.Arcade.Sdk/src/AssemblyResolution.cs]: And there is no binding redirect in msbuild.exe.config :(

Comment (tmat) [reply src/Microsoft.DotNet.Arcade.Sdk/src/AssemblyResolution.cs]: Which should probably be fixed.

File: src/Microsoft.DotNet.Arcade.Sdk/src/AssemblyResolution.cs [hunk 4]
```diff
@@ -21,13 +21,26 @@ private static Assembly AssemblyResolve(object sender, ResolveEventArgs args)
         {
             var name = new AssemblyName(args.Name);
 
-            if (name.Name.Equals("System.Collections.Immutable", StringComparison.OrdinalIgnoreCase))
+            if (!name.Name.Equals("System.Collections.Immutable", StringComparison.OrdinalIgnoreCase))
```
[the remark this answers is not in the document]
Comment (tmat) [reply src/Microsoft.DotNet.Arcade.Sdk/src/AssemblyResolution.cs]: Will do.

**Pull request merged**